src/algorithms/helpers/paths-diversity.ts

```diff
@@ -0,0 +1,36 @@
+import Path from '../pathfinders/path';
+
+function getDiversity(shortest: Path, current: Path): number {
+  const missingEdges = new Path();
+  for (const edge of current.pathData) {
+    if (shortest.pathData.indexOf(edge) === -1) {
+      missingEdges.pathData.push(edge);
+    }
+  }
+  const shortestLength = shortest.evaluate().getCost('distance');
+  const notSharedLength = missingEdges.evaluate().getCost('distance');
+
+  const diversity = notSharedLength / shortestLength;
+
+  return diversity > 1 ? 1 : diversity;
+ }
+
+export function printDiversities(paths: Path[]): void {
+  let maxDiversity = -1;
+  let minDiversity = 2;
+  let sumDiversity = 0;
+  for (const path of paths.slice(1)) {

```

Comment:
This would be nice place to map the list of paths into list of diversities and then reduce them